fix: esm

aprendendofelipe opened the pull request:
Resolve #1167
Resolve #1190

Shinigami92 commented:
All tests seem to pass 👀 
Could you add some explanation of what you did and why this solved some things?
Also can we somehow add a test if esm really works? Cause I still e.g. see `required` calls in the code.

aprendendofelipe commented:
My mistake! I opened the PR prematurely by accident (thought it was in my fork). I was still trying to reproduce the issue locally but couldn't with `npm link`, so I published to NPM to ensure the solution worked.

Changes made:
- Removed paths relative to `__dirname`.
- Removed `require`.
- Updated base for `templateFileName` to `node_modules/node-pg-migrate/templates`. This should be fine as `options.templateFileName` can be provided in exceptional cases.

What do you think?

aprendendofelipe commented:
Now we have problems with `.ts`

Shinigami92 commented:
> My mistake! I opened the PR prematurely by accident (thought it was in my fork). I was still trying to reproduce the issue locally but couldn't with `npm link`, so I published to NPM to ensure the solution worked.

You could try `file:../path-to-your-fork`, at least with pnpm that works

aprendendofelipe commented:
Everything working fine with `module.createRequire`.

Is it possible to test with `node-pg-migrate-exp`:

https://www.npmjs.com/package/node-pg-migrate-exp

aprendendofelipe commented:
Now without `__dirname` and `import.meta.url`, since we don't need to create `crossRequire` from a real path:

`createRequire(resolve('noop.js'))`

aprendendofelipe commented:
https://arethetypeswrong.github.io/?p=node-pg-migrate-exp%407.4.0-experimental-1189-7

[![arethetypeswrong.github.io/?p=node-pg-migrate-exp%407.4.0-experimental-1189-7](https://github.com/salsita/node-pg-migrate/assets/77860630/afc8374e-a915-4fa1-8b81-ba583b91bc5d)](arethetypeswrong.github.io/?p=node-pg-migrate-exp%407.4.0-experimental-1189-7)

Shinigami92 commented:
This was partially taken over in #1187

- #1187
